game/gamemusic.cpp

```diff
@@ -7,211 +7,351 @@
 #include "game/definitions/musicdefinitions.h"
 #include "dmusic/mixer.h"
 #include "resources.h"
+#include "dmusic.h"
 
 using namespace Tempest;
 
-struct GameMusic::MusicProducer : Tempest::SoundProducer {
-  MusicProducer():SoundProducer(44100,2){
+static constexpr uint16_t SAMPLE_RATE = 44100;
+
+struct GameMusic::MusicProvider : Tempest::SoundProducer {
+  using Tempest::SoundProducer::SoundProducer;
+
+  virtual void playTheme(const zenkit::IMusicTheme &theme, GameMusic::Tags tags) = 0;
+
+  virtual void stopTheme() = 0;
+
+  virtual void setEnabled(bool enable) = 0;
+
+  virtual bool isEnabled() = 0;
+
+  virtual const std::optional<zenkit::IMusicTheme> getPlayingTheme() = 0;
```

Comment:
generally in project I would like to move away from `getPlayingTheme` kind of call to simple `playingTheme`

Reply:
I was thinking about `MusicProvider` being an interface (the whole "polymorphism bad" shtick) so I was trying to avoid adding fields to it. Sure can do though if want me to.